service/src/test/kotlin/app/cash/backfila/service/persistence/BackfillRegisteredParametersTest.kt

```diff
@@ -0,0 +1,170 @@
+import misk.testing.MiskTest
+import misk.testing.MiskTestModule
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+
+@MiskTest(startService = true)
+class BackfillRegisteredParametersTest {
+  @Suppress("unused")
+  @MiskTestModule
+  val module: Module = BackfilaTestingModule()
+
+  @Inject
+  lateinit var configureServiceAction: ConfigureServiceAction
+
+  @Inject
+  lateinit var scope: ActionScope
+
+  @Inject
+  @BackfilaDb
+  lateinit var transacter: Transacter
+
+  @Inject
+  lateinit var queryFactory: Query.Factory
+
+  @Inject lateinit var backfila: Backfila
+
+  @Test
+  fun `add missing parameter`() {
+    scope.fakeCaller(service = "deep-fryer") {
+      configureServiceAction.configureService(
+        ConfigureServiceRequest.Builder()
+          .backfills(
+            listOf(
+              ConfigureServiceRequest.BackfillData(
+                "ChickenSandwich", "Description",
+                listOf(
+                  Parameter("name", "desc", false),
+                ),
+                null, null, false, null
+              )
+            )
+          )
+          .connector_type(Connectors.ENVOY)
+          .build()
+      )
+    }
+
+    backfila.createWetRun<BackfillRegisteredParameters>().execute()
+
+    transacter.transaction { session ->
+      val backfill = queryFactory.newQuery<RegisteredBackfillQuery>()
+        .uniqueResult(session)!!
+      assertThat(backfill.parameters).hasSize(1)
+      assertThat(backfill.parameters.single().name).isEqualTo("name")
+      assertThat(backfill.parameters.single().description).isEqualTo("desc")
+      assertThat(backfill.parameters.single().required).isFalse()
+      backfill.parameters.forEach { session.delete(it) }
+    }
+
+    backfila.createWetRun<BackfillRegisteredParameters>().execute()
+
+    transacter.transaction { session ->
+      val backfill = queryFactory.newQuery<RegisteredBackfillQuery>()
+        .uniqueResult(session)!!
+      assertThat(backfill.parameters).hasSize(1)
+      assertThat(backfill.parameters.single().name).isEqualTo("name")
+      assertThat(backfill.parameters.single().description).isEqualTo(null)
+      assertThat(backfill.parameters.single().required).isFalse()
+    }
+  }
+
+  @Test
+  fun `add missing multiple parameters`() {
+    scope.fakeCaller(service = "deep-fryer") {
+      configureServiceAction.configureService(
+        ConfigureServiceRequest.Builder()
+          .backfills(
+            listOf(
+              ConfigureServiceRequest.BackfillData(
+                "ChickenSandwich", "Description",
+                listOf(
+                  Parameter("abc", "desc1", false),
+                  Parameter("def", "desc2", false),
+                ),
+                null, null, false, null
+              )
+            )
+          )
+          .connector_type(Connectors.ENVOY)
+          .build()
+      )
+    }
+
+    backfila.createWetRun<BackfillRegisteredParameters>().execute()
```

Comment:
Do you need this one? I think not?

Reply:
it should do nothing - so if it were buggy, it would fail trying to duplicate insert